### PR TITLE
Add st2chatops support

### LINKED ISSUE #206: st2chatops missing
Hi team.

I believe that st2chatops is the standard package on ST2 but docker-compose.yml does not include st2chatops image and configuration.

Does not have a plan to deploy st2chatops ?

### PR DESCRIPTION
**Pull request by stevemuskiewicz**
This is my initial attempt at adding chatops support for #206 

Closes #206

It has been minimally tested using the Slack adapter and an st2 API key for authentication from the chatops container to the st2 api.

I included a `st2chatops-startup.sh` that uses `curl` to repeatedly check that the st2api is up and running before starting hubot since I was having problems with the chatops container starting before the API and then it never read any of the aliases configured in st2 (there may be a better way to achieve this)

The `files/st2chatops.env` was taken directly from https://github.com/StackStorm/st2chatops/blob/master/st2chatops.env

### CODE REVIEW

**Comment by AlexRRR** [comment]
Looks pretty good. One concern would be that if the docker-compose is started without setting the `HUBOT_SLACK_TOKEN` or configuring the file the container complains over and over again with:

```
st2chatops_1           | error: Response not OK:  internal_error
st2chatops_1           | info: Unable to rtm.start, attempting reconnect
st2chatops_1           | warn: Reconnecting, on attempt 140
st2chatops_1           | info: Unable to rtm.start, attempting reconnect
st2chatops_1           | warn: Reconnecting, on attempt 141
st2chatops_1           | error: Response not OK:  internal_error
```
As there might be people who don't want st2chatops enabled it should be off by default. Not sure if another envvar could be an option.

**Comment by stevemuskiewicz** [comment]
@AlexRRR yeah great point, I hadn't considered that but def should make chatops optional.  Lemme poke at it and see what I can come up with.  envvar does seem like the way to go...

**Comment by stevemuskiewicz** [comment]
OK I think I'm going to implement this by putting the chatops service in a separate docker compose file along the lines of this https://stackoverflow.com/a/61107015/2852180

(Don't seem to be many other options for defining "optional" services in a single compose file)

File: scripts/st2chatops-startup.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# (minimal) check for chatops env
+if [[ -z "$HUBOT_ADAPTER" ]] ; then
+  echo "WARNING: \$HUBOT_ADAPTER is not set, cannot start hubot..."
+  # just sleep indefinitely to avoid this container getting repeatedly restarted
```
Comment (armab) [inline scripts/st2chatops-startup.sh:6]: Having the container running when in fact it had incorrect configuration may bring confusion in terms of liveness and understanding the service health. Failing fast and hard from the misconfigured container would be expected behavior to me.

On top of that, making users configure the restart policy via docker-compose is more preferred:
https://docs.docker.com/compose/compose-file/compose-file-v3/#restart

**Review by armab — CHANGES_REQUESTED** [review]
That's a great addition :+1: 
Thanks a lot for the contribution!

I left a couple of comments which would even simplify things a bit.
Please also include the Changelog record for this feature.
Besides of that, looks good!

**Comment by armab** [comment]
I get an idea about splitting into 2 docker-compose files, but that would complicate the configuration and experience for the users. We learned it, the hard way, and I'd prefer simplicity first for the default configuration.

To avoid an infinite restart loop, what if we'll do like we did for the default VM StackStorm installation with the services?
This means try to start the st2chatops container a few times (or one time), but fail and stop immediately on misconfiguration.

Here are a few ideas:

> `restart: always` will restart the container regardless of the exit code, so even if exit code of the process running inside the container is `0`. `restart: on-failure` restarts the container on a non zero exit code of the process.

That way we could control the st2chatops with `ST2_CHATOPS_ENABLE: 0` as a docker-compose ENV variable and `exit 0` in `st2chatops-startup.sh`.

Alternatively, looks like there is a feature `restart: on-failure:10` (per https://stackoverflow.com/a/43006595/4533625) which would allow restarting the st2chatops container 10 times on failure before giving up.

**Comment by stevemuskiewicz** [comment]
@armab Agree completely about simplicity and keeping the single compose file - I'm definitely not thrilled with the multiple file approach.  I'll take a look at your suggested options and try to come up with something simpler using the single compose file.

**Comment by stevemuskiewicz** [comment]
@armab i went with your `restart: on-failure:<count>` suggestion although I did discover that hubot (at least with the slack adapter) seems to want to repeatedly try to restart itself if it's not configured properly (ie. we give it the bogus "xoxb-CHANGE-ME-PLEASE" placeholder value) in a way that doesn't play well with the compose file.  So I just opted to default it to an empty value, which at least gives something useful in the `st2chatops` service log ("No token provided to Hubot")

Not as ideal as you had suggested with it failing when not configured but it was the best I could figure out for now.  

I did actually try messing with setting `HUBOT_SLACK_RTM_CLIENT_OPTS` to control "retryConfig" per https://slack.dev/hubot-slack/advanced_usage#customizing-rtmstart-options but either I'm misunderstanding what that does or else it doesn't work right so I just gave up on that 🤷

**Review by armab — APPROVED** [review]
Looks great! :+1: 

Thanks a lot for the contribution!

**Pull request merged**